x/gov/client/cli/parse.go

```diff
@@ -10,15 +10,23 @@ import (
 	govClientUtils "github.com/cosmos/cosmos-sdk/x/gov/client/utils"
 )
 
+func readProposalType() string {
+	return govClientUtils.NormalizeProposalType(viper.GetString(flagProposalType))
+}
+
 func parseSubmitProposalFlags() (*proposal, error) {
 	proposal := &proposal{}
 	proposalFile := viper.GetString(flagProposal)
 
 	if proposalFile == "" {
 		proposal.Title = viper.GetString(flagTitle)
 		proposal.Description = viper.GetString(flagDescription)
-		proposal.Type = govClientUtils.NormalizeProposalType(viper.GetString(flagProposalType))
+		proposal.Type = readProposalType()
```

Comment:
Just curious to learn why? Wasn't the inline call enough? I can't find any other place where `readProposalType()` is used